internal/manifests/manifests.go

```diff
@@ -0,0 +1,173 @@
+package manifests
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/filemiddleware"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/openshift/assisted-service/pkg/s3wrapper"
+	"github.com/openshift/assisted-service/restapi"
+	operations "github.com/openshift/assisted-service/restapi/operations/manifests"
+	"github.com/sirupsen/logrus"
+)
+
+var _ restapi.ManifestsAPI = &Manifests{}
+
+const manifestFolder = "manifests"
+
+// NewManifestsAPI returns manifests API
+func NewManifestsAPI(db *gorm.DB, log logrus.FieldLogger, objectHandler s3wrapper.API) *Manifests {
+	return &Manifests{
+		db:            db,
+		log:           log,
+		objectHandler: objectHandler,
+	}
+}
+
+// Manifests represents manifests handler implementation
+type Manifests struct {
+	db            *gorm.DB
+	log           logrus.FieldLogger
+	objectHandler s3wrapper.API
+}
+
+func (m *Manifests) CreateClusterManifest(ctx context.Context, params operations.CreateClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, m.log)
+	log.Infof("Creating manifest in cluster %s", params.ClusterID)
+
+	if params.CreateManifestParams.Folder == nil {
+		defaultFolder := models.CreateManifestParamsFolderManifests
+		params.CreateManifestParams.Folder = &defaultFolder
+	}
+
+	cluster, apierr := common.GetCluster(ctx, m.log, m.db, params.ClusterID.String())
+	if apierr != nil {
+		return common.GenerateErrorResponder(apierr)
+	}
+
+	fileName := filepath.Join(*params.CreateManifestParams.Folder, *params.CreateManifestParams.FileName)
+	manifestContent, err := base64.StdEncoding.DecodeString(*params.CreateManifestParams.Content)
+	if err != nil {
+		log.Errorf("Cluster manifest %s for cluster %s failed to base64 decode: [%s]",
+			fileName, cluster.ID, *params.CreateManifestParams.Content)
+		return common.GenerateErrorResponder(common.NewApiError(http.StatusBadRequest,
+			errors.New("failed to base64-decode cluster manifest content")))
+	}
+
+	objectName := GetManifestObjectName(*cluster.ID, fileName)
+	if err := m.objectHandler.Upload(ctx, manifestContent, objectName); err != nil {
+		return common.GenerateErrorResponder(common.NewApiError(http.StatusInternalServerError,
```

Comment:
No need to both `common.GenerateErrorResponder` and `common.NewApiError` you can just use `common.NewApiError` - relevant to all the errors that use both.